stdlib/array.ml

```diff
@@ -63,6 +63,20 @@ let make_matrix sx sy init =
   done;
   res
 
+let init_matrix sx sy f =
+  if sy < 0 then invalid_arg "Array.init_matrix";
+  let res = create sx [||] in
+  if sy > 0 then begin
```

Comment:
This optimization (avoiding the loop if `sy = 0`) is not present in `make_matrix`. Not opposed to it, but I would rather have code with the same shape in both `init_matrix` and `make_matrix`.

Reply:
It’s not just an optimization: without this test, we would evaluate `f x 0` for all `x` even if `sy = 0`. This is the meaning of the comment about #6575 in `init`. I made that comment more explicit, and added a new one in `init_matrix`.

I aligned the implementation of `make_matrix` with that of `init_matrix` as you suggested.